webpack.config.js

```diff
@@ -0,0 +1,196 @@
+const webpack = require('webpack');
+const conf = require('./tasks/config.js');
+const pkg = require('./package.json');
+const path = require('path');
+
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+// Doing a webpack hack for es6: https://github.com/webpack-contrib/uglifyjs-webpack-plugin/issues/33
+const UglifyJsPlugin = require('uglifyjs-webpack-plugin');
+const FailPlugin = require('webpack-fail-plugin');
+const ExtractTextPlugin = require('extract-text-webpack-plugin');
+const autoprefixer = require('autoprefixer');
+
+// Our current environment
+let env = {};
+
+// Our environment aliases
+const ENV_ALIAS = {
+  PROD: [
+    'production',
+    'prod'
+  ],
+  DEV: [
+    'development',
+    'dev'
+  ],
+  TEST: [
+    'test'
+  ]
+};
+
+/**
+* Function to return if the environment matched the environment alias
+*/
+function isEnv(envAlias) {
+  // Check for default case
+  if (envAlias === ENV_ALIAS.DEV && (!env || Object.keys(env).length === 0)) {
+    return true;
+  }
+
+  return envAlias.some(alias => {
+    if (env[alias]) {
+      return true;
+    }
+    return false;
+  });
+}
+
+function isNotEnv(envAlias) {
+  return !isEnv(envAlias);
+}
+
+module.exports = function (webpackEnv) {
+  // Set our environemnt
+  env = webpackEnv;
+  // Create our base webpack configuration
+  const webpackConf = {
+    module: {
+      loaders: [
+        {
+          test: /\.json$/,
+          loaders: [
+            'json-loader'
+          ]
+        },
+        {
+          test: /\.js$/,
+          exclude: /node_modules/,
+          loader: 'eslint-loader',
+          enforce: 'pre'
+        },
+        {
+          test: /\.js$/,
+          exclude: /node_modules/,
+          loaders: [
+            'babel-loader'
+          ]
+        }
+      ]
+    },
+    plugins: [],
+    output: {},
+    entry: []
+  };
+
+  // LOADERS
+  if (isEnv(ENV_ALIAS.PROD)) {
```

Comment:
I agree that it is a little harder to read. But having multiple files with slight changes is just as bad in my opinion. the `--env` is still relatively new, From what I can tell was introduced about a year ago, which is why you will not see a lot of webpack configs following the pattern because migrating can be a pain.

I'm open to either way at this point, but I personally feel a single file will be more maintainable